src/OptimalTransport.jl

```diff
@@ -784,4 +786,103 @@ function quadreg(mu, nu, C, ϵ; θ=0.1, tol=1e-5, maxiter=50, κ=0.5, δ=1e-5)
     return sparse(γ')
 end
 
+"""
+    ot_cost(
+        c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution; plan=nothing
+    )
+
+Compute the optimal transport cost for the Monge-Kantorovich problem with univariate
+distributions `μ` and `ν` as source and target marginals and cost function `c` of
+the form ``c(x, y) = h(|x - y|)`` where ``h`` is a convex function.
+
+In this setting, the optimal transport cost can be computed as
+```math
+\\int_0^1 c(F_\\mu^{-1}(x), F_\\nu^{-1}(x)) \\mathrm{d}x
+```
+where ``F_\\mu^{-1}`` and ``F_\\nu^{-1}`` are the quantile functions of `μ` and `ν`,
+respectively.
+
+A pre-computed optimal transport `plan` may be provided.
+
+See also: [`ot_plan`](@ref), [`emd2`](@ref)
+"""
+function ot_cost(
+    c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution; plan=nothing
+)
+    cost, _ = if plan === nothing
+        quadgk(0, 1) do q
+            return c(quantile(μ, q), quantile(ν, q))
+        end
+    else
+        quadgk(0, 1) do q
+            x = quantile(μ, q)
+            return c(x, plan(x))
+        end
+    end
+    return cost
+end
+
+"""
+    ot_plan(c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution)
+
+Compute the optimal transport plan for the Monge-Kantorovich problem with univariate
+distributions `μ` and `ν` as source and target marginals and cost function `c` of
+the form ``c(x, y) = h(|x - y|)`` where ``h`` is a convex function.
+
+In this setting, the optimal transport plan is the Monge map
+```math
+T = F_\\nu^{-1} \\circ F_\\mu
+```
+where ``F_\\mu`` is the cumulative distribution function of `μ` and ``F_\\nu^{-1}`` is the
+quantile function of `ν`.
+
+See also: [`ot_cost`](@ref), [`emd`](@ref)
+"""
+function ot_plan(c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution)
+    # Use T instead of γ to indicate that this is a Monge map.
+    T(x) = quantile(ν, cdf(μ, x))
+    return T
+end
+
+"""
+    ot_cost(c::SqEuclidean, μ::Union{Normal,MvNormal}, ν::Union{Normal,MvNormal})
```

Comment:
This is not correct (also the implementation below) since it would allow combinations of `Normal` and `MvNormal` of arbitrary dimensions.

Reply:
```suggestion
    ot_cost(::SqEuclidean, μ::MvNormal, ν::MvNormal)
```

Reply:
> This is not correct (also the implementation below) since it would allow combinations of `Normal` and `MvNormal` of arbitrary dimensions.

You mean for the compiler? Cause when I try to use mu as Normal and nu as MvNormal, the code throws an error. The same if I try MvNormals with different dimensions, which should also not be allowed.

Reply:
No, this is just a lucky coincidence. The code will throw an error for both mu and nu as `Normal` as well since `Normal` does not have a field `\Sigma` (and probably the functions such as `sqbures` are not defined for real numbers). If such problems would be fixed, then (probably) it would not error anymore.

In any case, it seems easier to disallow these combinations completely by specifying a correct function signature of `ot_cost` instead of relying on some other errors in some functions that are called by it. This would also be clearer for users since the error message is easier to understand.

Reply:
I see. Ok, I'll fix this and resubmit.

Reply:
The code actually works for mu and nu as Normals, but my guess is that it's actually sending the to the 1D implementation.